### DISCUSSION TITLE
Railsメジャーアップデートのツラミについて

### DISCUSSION THREAD

**Discussion by danimal141** [roppongirb/discussions#45] [General]

**Comment by hatappi** [comment]
## 概要
メジャーアップデートしてて何か辛い経験したことある？

**Comment by hatappi** [comment]
## アップグレード時のアレコレ
- Rails 2 -> xxx
  - xxxx
- Rails 4 -> 5
  - 既存のテストが落ちてひたすら治す
  - Gem 自体のアップデートも必要だったので、大きいものからアップデートしていき潰していく
  - Rails4の時代がながいので CI 上とかで warning が出るのを潰していく
    - 一気に上げずに細かくできたものから上げていく
- Rails 5 -> 6
  - switch_points が使えなくなるので Rails6 から入ったデフォルトを使うようにマイグレーション対応
- Rails 6 -> 7
  - 苦労中

## 共通
- Gem でバージョンが固定されていると詰む
- そもそもテストを書いてかないと。。。そもそもこのテスト大丈夫だっけなど。。。
- 一大プロジェクトになりがち

## 番外編
- リリースは勇気がいるぜ！
  - テスト自体は通っても後日問題に気づくことも
- Ruby 2.6 とかだと使用している gem によってエラーが出たりして Rails のアップデートをする必要があることも
- 取引先とかの Rails プロジェクトをメンテしている時にどう説明するのか
  - アップデートしないことによる問題を提示すると良さそう👍 
- Java 6 がサポートされなくなるのでそのアップデートをやる

**Reply by hatappi** [reply]
Rails 7 の JS 周りどうするの。。。。
https://world.hey.com/dhh/rails-7-will-have-three-great-answers-to-javascript-in-2021-8d68191b

**Comment by hatappi** [comment]
## アップグレード方法
- https://railsguides.jp/upgrading_ruby_on_rails.html
- プロダクト固有の問題とかもあるのでアップグレード時の手順だったりハマりどころが記載されているとそれを参考にしつつ対応
- Rails guide もすべてではないので、その時は都度調べて対応
  - 本当にわからん時は Rails のバージョン戻しつつ問題がおきるところ特定してソースコード見たりとかも
- Changelog は多すぎて全部おいきれてない https://github.com/rails/rails/releases/tag/v7.0.0

**Reply by hatappi** [reply]
Changelog の各変更に対して PR だったり commit のリンクがなくて深ぼって変更を見れない問題

**Comment by hatappi** [comment]
Rails 7 の JS 周りどうするの。。。。
https://world.hey.com/dhh/rails-7-will-have-three-great-answers-to-javascript-in-2021-8d68191b

**Comment by hatappi** [comment]
## gem のアップデートが Rails のアップデートのブロッカーになってる場合
- 基本待つ
- gem とかで更新頻度が低いものの時に待ちになるとつらそう。。。
